crank/tests/test_stack.py

```diff
@@ -0,0 +1,187 @@
+"""
+Unit and regression test for the crank package.
+"""
+
+import pytest
+import os, sys, json
+import numpy as np
+from crank.DihedralScanner import DihedralScanner, Molecule
+from crank.QMEngine import QMEngine
+import geometric
+from geometric.nifty import bohr2ang
+
+try:
+    import qcengine
+    import psi4
+except:
+    pass
+
+class Psi4QCEngineEngine(QMEngine):
+    def __init__(self, *args, **kwargs):
+        super(Psi4QCEngineEngine, self).__init__(*args, **kwargs)
+        self.stored_results = dict()
+
+    def load_input(self, input_file):
+        """ Load molecule from an xyz file """
+        self.M = Molecule(input_file)
+
+    def optimize_geomeTRIC(self):
+        """ run the constrained optimization using geomeTRIC package:
+        1. Create a json input for a constrained optimization
+        2. Run the job
+        """
+        # step 1
+        in_json_dict = self.create_in_json_dict()
+        # step 2
+        key = os.getcwd()
+        self.stored_results[key] = geometric.run_json.geometric_run_json(in_json_dict)
+
+    def create_in_json_dict(self):
+        constraints_string = "$set\n"
+        for d1, d2, d3, d4, v in self.dihedral_idx_values:
+            # geomeTRIC use atomic index starting from 1
+            constraints_string += "dihedral %d %d %d %d %f\n" % (d1+1, d2+1, d3+1, d4+1, v)
```

Comment:
This relies on geomeTRIC to be able to parse a dictionary for constraints. This will require designing a new interface for the constraints in geomeTRIC first, which needs its own testing and lots of other work. For now I think the safest way is to use the same format as before. Later when that feature become available, updating the interface here will be trivial.